server/src/Commands.def

```diff
@@ -12,7 +12,9 @@
 #error Please define the 'X(id, command, description)' macro before inclusion!
 #endif
 
-X(Candidates, "candidates", "print completion candidates (require previous complete)")
+X(Candidates, "candidates",
+  "PREFIX STYLE - print completion candidates (require previous complete). "
```

Comment:
I think, for irony.el/irony-completion.el and irony-server, I would prefer that you use consistently the term "style" for this candidate argument.

The style should be a list of the following values:

- in Elisp: `'exact`, `'case-insensitive` and `'smart-case`
- in C++ use an enum class for std::string caseStyle, so you benefit more from the type system, only when read/validate the command you deal with a string

I prefer the term style, because in the future I can see something not related to case, which would be fuzzy search.
It could be a new function, but if we have an arbitrary style value here, it is sufficient.

On the other hand, you change in `company-irony-ignore-case` is perfect, very sensible choice to keep compatibility,
but I think irony-server sides this could be a bit clearer and leaves the door open for the future.